test/src/test_derive.rs

```diff
@@ -178,6 +192,236 @@ fn test_derive_owned_to_variant() -> bool {
     ok
 }
 
+#[derive(NativeClass)]
+#[inherit(Reference)]
+struct MinimalDerive(i64);
+
+#[methods]
+impl MinimalDerive {
+    fn new(_owner: &Reference) -> Self {
+        Self(54)
+    }
+
+    #[export]
+    fn answer(&self, _owner: &Reference) -> i64 {
+        self.0
+    }
+}
+
+fn test_derive_nativeclass() -> bool {
+    println!(" -- test_derive_nativeclass");
+
+    let ok = std::panic::catch_unwind(|| {
+        let thing = Instance::<MinimalDerive, _>::new();
+        let base = thing.into_base();
```

Comment:
I wonder if we should explictly specify the type of variable `base` here?
That would additionally test that `Reference` is really chosen as the base, and not e.g. `Object`.

Same for other tests.

Reply:
I tried to do it.
```rust
let thing = Instance::<_, _>::new();
```

```
error[E0282]: type annotations needed
   --> test\src\test_derive.rs:217:34
    |
217 |         assert_eq!(unsafe { base.call("answer", &[]).to::<i64>() }, Some(54));
    |                                  ^^^^ cannot infer type
    |
    = note: type must be known at this point

error[E0599]: no method named `call` found for struct `gdnative::prelude::Ref<_, gdnative::prelude::Unique>` in the current scope
   --> test\src\test_derive.rs:217:34
    |
217 |         assert_eq!(unsafe { base.call("answer", &[]).to::<i64>() }, Some(54));
    |                                  ^^^^ method not found in `gdnative::prelude::Ref<_, gdnative::prelude::Unique>`
```

Reply:
It meant for the `base` variable, it should probably be:
```rs
let base: Ref<MinimalDerive, Unique> = thing.into_base();
```

Reply:
I'm sorry. I did not understand.

I'll do that for now, but, I think it is `trybuild` oriented test case because the test is to see if the macro causes compile errors.